rhizome/host/lib/slice_setup.rb

```diff
@@ -47,4 +48,15 @@ def start_systemd_unit
     r "systemctl start #{@slice_name}"
     r "echo \"root\" > /sys/fs/cgroup/#{@slice_name}/cpuset.cpus.partition"
   end
+
+  def valid_cpuset?(str)
+    str.split(",").all? do |part|
+      if part.include?("-")
+        r = part.split("-")
+        r.size == 2 && r.all? { _1.to_i.to_s == _1 } && r[0].to_i <= r[1].to_i
```

Comment:
You sure about .to_i? If it gets any weird input, it'll give you zero, rather than raising. `Integer(x, 10)` is the most restrictive.

Reply:
Ah okay, this is among the shortest ways to test for reflexivity of the string representation plus inequality. Very slick, @jeremyevans you got something that's clearer (and maybe avoids some extra to_i conversion) but not longer? If not, I think its fine to keep.

Reply:
3 characters shorter, and avoids duplicate `to_i` calls:

```ruby
r.size == 2 && r == (s = r.map(&:to_i)).map(&:to_s) && s[0] <= s[1]
```

Not sure that is clearer.  Maybe if you move the `s` assignment to the line above, but that's technically longer if you count the whitespace (5 characters shorter without the whitespace, since your can avoid the parentheses in that case).

I'm not requesting the code be changed, what is there seems fine.